lib/midcom/helper/imagepopup/handler/upload.php

```diff
@@ -0,0 +1,181 @@
+                if(in_array($_SERVER['HTTP_ORIGIN'], $accepted_origin))
+                {
+                    midcom::get()->header('Access-Control-Allow-Origin: ' . $_SERVER['HTTP_ORIGIN']);
+                } 
+                else
+                {
+                    throw new midcom_error_forbidden("HTTP/1.0 403 Origin Denied");
+                }
+            }
+            
+            if(!in_array(strtolower(pathinfo($temp['name'], PATHINFO_EXTENSION)), array("gif", "jpg", "png")))
+            {
+                throw new midcom_error_notfound("HTTP/1.0 500 Invalid extension.");
+            }
+        }
+        
+        // Get the data
+        $temp_name = $temp['tmp_name'];
+        $mimetype = $temp['type'];
+        $parentguid = "";
+        
+        // Do we have guid ?
+        switch($handler_id)
+        {
+            case '____ais-imagepopup-upload_image_noobject':
+                $parentguid = $this->_topic->guid;
+                break;
+            case '____ais-imagepopup-upload_image':
+                $parentguid = $args[0];
+                break;
+        }
+        
+        // Set modified filename
+        $filename = $this->get_modify_filename($temp['name']);
+        
+        // Insert the image into database
+        $attachment = $this->insert_database($filename, $mimetype, $parentguid);
+        $data['attachment'] = $attachment;
+        
+        // Get target to write the file
+        $target = $this->get_data_from_database($filename, $parentguid);
+        
+        // Write the file
+        $this->write_the_file($temp_name, $target);
+        
+        // Make a response for editor.uploadImages() function
+        $location = midcom_db_attachment::get_url($attachment);
+        $data['location'] = $location;
+        
+        // Return image location as JSON
+        return new midcom_response_json(array('location' => $location));
+    }
+    
+    /**
+     * @param string $filename The file name
+     * @return string The modified file name
+     */
+    private function get_modify_filename($filename)
+    {
+        $filename = midcom_db_attachment::safe_filename($filename);
+        $pieces = explode('.', $filename);
+        $core = "modified";
+        $extension = array_pop($pieces);
+        $time = $this->get_current_time('Y-m-d-H-i-s');
+        $modifyFilename = $core . $time . "." . $extension;
+        
+        return $modifyFilename;
+    }
+    
+    /**
+     * @param string $pattern The pattern for date
+     * @return date The date for modified file name
+     */
+    private function get_current_time($pattern)
+    {
+        return date($pattern, time());
+    }
+    
+    /**
+     * @param string $filename The file name
+     * @param string $parentguid The parent GUID
+     * @return string The file destination
+     */
+    private function get_data_from_database($filename, $parentguid)
+    {
+        $filename = midcom_db_attachment::safe_filename($filename);
```

Comment:
`safe_filename` was already called in `get_modify_filename`, so calling it again here is redundant